thunder/torch/__init__.py

```diff
@@ -4733,10 +4736,12 @@ def embedding(
     return reshape(flatten_output, output_shape)
 
 
+# TODO Once we have fusible index_put we can implement it using primitives
+# For now we just use the PyTorch implementation
 @torchsymbol(torch.ops.aten.embedding_backward)
 def embedding_backward(grad, indices, num_weights, padding_idx, scale_grad_by_freq, sparse):
-    result = prims.embedding_backward(grad, indices, num_weights, padding_idx, scale_grad_by_freq, sparse)
-    return result
+    shape = (num_weights, grad.shape[-1])
+    return TensorProxy(shape=shape, device=grad.device, dtype=grad.dtype)
```

Comment:
Same question here about directly constructing and return a TensorProxy over calling some op that would create one

I guess I'm a little confused about operations in the torch namespace directly constructing TensorProxies, as that seems more like a primitives thing? Curious to hear your thoughts

Reply:
We started Thunder thinking that the terminal level should be the `thunder.prims` namespace. It was before moving to the multi-level representation of PyTorch operators, there was just a jump from PyTorch straight to prims, I think. We also didn't have a direct lowering from `thunder.torch` to `torch` at the time. In similar cases, like this indirection is unnecessary, and instead of forming proxies in the meta function of `prims.embedding_backward` we can do that directly in `thunder.torch.embedding_backward`. The subsymbols of this operation would be empty and this is the indicator to treat it as a primitive both for execution and for transformations.

Reply:
My guess is that this approach will work today but will require revisiting in the future. If we start to use the primitive language context as a signal that we should do some special account for shape constraints, for example (fyi @jjsjann123)

Reply:
Effectively this is expanding the set of `terminal level symbols` to be `prims` + `symbols without prim decomposition`?

I see the benefit of eliminating one level of indirection.
i.e. effectively this PR just lifts (or lowers) these torch.nn.functional.symbols to be our `expanded prims`. I don't have a strong opinion against it.

> My guess is that this approach will work today but will require revisiting in the future.

I'm curious what's the concern you have @mruberry ? Are you just worried about this making it too easy to smuggle `effective prims` into the code base and it's hard to police that?